sources/packages/api/src/route-controllers/models/common.dto.ts

```diff
@@ -0,0 +1,78 @@
+import { IsNotEmpty, IsOptional } from "class-validator";
+
+/**
+ * TODO: Address DTOs have been with 2 different names as the
+ * TODO: property province is used as provinceState in some places in code.
+ **/
+/**
+ * Common DTO for Address
+ ** This DTO is used in API request body.
+ */
+export class AddressInDTO {
+  @IsNotEmpty()
+  addressLine1: string;
+  @IsOptional()
+  addressLine2: string;
+  @IsNotEmpty()
+  city: string;
+  @IsNotEmpty()
+  provinceState: string;
+  @IsNotEmpty()
+  country: string;
+  @IsNotEmpty()
+  postalCode: string;
+}
+
+/**
+ * Common DTO for Address
+ ** This DTO is used in API request body.
+ */
+export class AddressInfoInDTO {
+  @IsNotEmpty()
+  addressLine1: string;
+  @IsOptional()
+  addressLine2: string;
+  @IsNotEmpty()
+  city: string;
+  @IsNotEmpty()
+  province: string;
+  @IsNotEmpty()
+  country: string;
+  @IsNotEmpty()
+  postalCode: string;
+}
+
+/**
+ * Common DTO for Address.
+ ** This DTO is returned as API response body.
+ */
+export class AddressOutDTO {
```

Comment:
does this need `Out`? IMO, it looks like a generic address interface/class, which can also be used in/as a non-DTO interface or classes

Reply:
But if this part of a InDTO then we need class validators in it for validation. So created DTO with In and Out where out is without class validators.

Reply:
no, iam not talking about in DTO, iam  talking about other nondto interfaces

Reply:
Yes at non-dto level also there should be a generic interface created to represent address.